api/v1beta1/glance_webhook.go

```diff
@@ -88,9 +130,19 @@ func (r *Glance) ValidateUpdate(old runtime.Object) error {
 	// Type can either be "split" or "single": we do not support changing layout
 	// because there's no logic in the operator to scale down the existing statefulset
 	// and scale up the new one, hence updating the Spec.GlanceAPI.Type is not supported
-	o := old.(*Glance);
-	if (r.Spec.GlanceAPI.Type != o.Spec.GlanceAPI.Type) {
-		return errors.New("GlanceAPI deployment layout can't be updated")
+	o := old.(*Glance)
+	for key, glanceAPI := range r.Spec.GlanceAPIs {
+		// When a new entry (new glanceAPI instance) is added in the main CR, it's
+		// possible that the old CR used to compare the new map had no entry with
+		// the same name. This represent a valid use case and we shouldn't prevent
+		// to grow the deployment
+		if o.Spec.GlanceAPIs[key].Type == "" {
```

Comment:
When I first read this line I was expecting the comment above it to mention something about `Type`, but I think the comment makes it clear that we care whether the `GlanceAPI` existed at all in the old `Glance`.  But maybe it would be better to use a check like `if _, ok := o.Spec.GlanceAPIs[key]; ok {` to more clearly-align with what the comment is indicating?  I'm truly just asking, not suggesting that it be one way or the other.

Reply:
Yes, I think it makes sense: instead of checking a potential `empty` parameter in an `entry` that might have not been there in the `old` object provided by the runtime, we can just use the way you proposed (thanks, it's also more elegant in terms of style), except for the fact that should be in the form `if _, found := o.Spec.GlanceAPIs[key]; !found {`, (with the `not` to indicate that it's not found) so we can move forward with the iteration.
In my local test I get:

```
[stack@osp-storage-04 layout]$ oc edit glance glance
Edit cancelled, no changes made.
``` 
when I try to add a new instance, so I think it worked as expected. 
Thanks for this suggestion!